src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -503,7 +503,7 @@ private static async Task<bool> IsHubMethodAuthorizedSlow(IServiceProvider provi
             // AuthorizationPolicy.CombineAsync only returns null if there are no policies and we check that above
             Debug.Assert(authorizePolicy != null);
 
-            var authorizationResult = await authService.AuthorizeAsync(principal, authorizePolicy);
+            var authorizationResult = await authService.AuthorizeAsync(principal, resource, authorizePolicy);
```

Comment:
`resource` is an `object` though right? So this will just box it anyway.

Reply:
That is correct :(

Reply:
Can I get to the HttpContext from here?

Reply:
I want to use the query string from the original negotiate request, which I think its fair game?

Reply:
>Can I get to the HttpContext

Yes.

Reply:
The resource gives you `HubCallerContext` which has a `.GetHttpContext` extension method.

Reply:
Is there some standard way that IAuthorizationHandler can use an HttpContext today? Is the `resource` normally the HttpContext itself?

Reply:
> Is there some standard way that IAuthorizationHandler can use an HttpContext today?

Not unless the handler uses `IHttpContextAccessor`.

> Is the resource normally the HttpContext itself?

Depends on what best describes the resource being authenticated. In MVC it's an [`AuthorizationFilterContext`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.filters.authorizationfiltercontext?view=aspnetcore-2.2) which includes `HttpContext` but also the resolved action information.